### PR TITLE
Embed 'alias' functionality into 'vendor\bin' and Load User Lua files

### LINKED ISSUE #1862: 1.3.6: Error on start path with trailing slash in quotes.
### Purpose of the issue
- [x] Bug report (encountered problems/errors)
- [x] Question

### Description of the issue 

Not sure if bug or question, so both.

I use Total Commander to launch Cmder in a specific directory, with the following pattern:
`cmder.exe /start "%p"`

%p represents the path of the source directory in TC, and will be sent to the process. The quotes are needed if spaces are present.

Since upgrading from 1.3.4 to 1.3.6 this does no longer work, and I get the following error messages:

clipboard

According to the [README](https://github.com/cmderdev/cmder/blob/master/README.md), this should also be a valid argument

> /START [start_path]  |  Folder path to start in.

I have found some other issues giving the impression that now Cmder starts in the "current working directory", or that the correct argument should be maybe `/dir` but this does not work either.

In 1.3.4 the help dialog looked like this:

image

The help dialog in 1.3.6 is a mess to be honest, in addition to the glaring typo in the first message. Why are there [] around the argument, and why a OR ( | ) with the same on both sides? It's very confusing.

I have also found the following:
- When launch with a trailing \ the above error appears, then the usage dialog.
`Cmder.exe /start "g:\download\"`

- Without a trailing slash, Cmder launches, but only in %userprofile%.
`Cmder.exe /start "g:\download"`

### PR DESCRIPTION
**Pull request by daxgames**
# Fix #1862 - 1.3.6: Error on start path with trailing slash in quotes.

# Enable/Disable Cmder features by detected comspec

* Add `vendor\lib\lib_base` function `cmder_shell` that sets 3 new env vars"
  * `CMDER_SHELL` = [cmd|tcc|tccle]
  * `CMDER_CLINK` = [1|0] Default = 1 or On
  * `CMDER_ALIASES` = [1|0] Default = 1 or On

* `vendor\init.bat` uses the new variables to determine whether to configure/load certain features at startup.

# Move bin\alias.bat to vendor\bin\alias.cmd

* Move `%CMDER_ROOT%\config\user-aliases.bat` to `%CMDER_ROOT%\config\user_aliases.cmd` related to #1870, #1806, and #1675
* Move `bin\alias.bat` to `vendor\bin\alias.cmd` to embed its functionality into cmder so `%CMDER_ROOT%\bin` becomes a completely user controlled folder like `%CMDER_ROOT%\config`
* Adds `%CMDER_ROOT%\bin;%CMDER_ROOT%\vendor\bin` to the path.
* Starting cmder with the old `%CMDER_ROOT%\bin\alias.bat` and `%CMDER_ROOT%\vendor\bin\alias.cmd` displays the following message:

```
Cmder's 'alias' command has been moved into '%CMDER_ROOT%\vendor\bin\alias.cmd'
to get rid of this message either:

Delete the file '%CMDER_ROOT%\bin\alias.bat'

or

If you have customized it and want to continue using it instead of the included version
  * Rename '%CMDER_ROOT%\bin\alias.bat' to '%CMDER_ROOT%\bin\alias.cmd'.
  * Search for 'user-aliases' and replace it with 'user_aliases'. 
)
```

# Loads user *.lua files after `%cmder_root%/vendor/clink.lua`

* Loads *.lua from the following sources in order:
  * `%cmder_root%/vendor/clink.lua`
    * `%cmder_root%/vendor/clink-completions/*.lua`
    * `%cmder_root%/config/*.lua`
    * `[cmder_user_config_root]/*.lua` If specified at launch using `-c [cmder_user_config_root]`

* Allows user more control over clink settings.  `%cmder_root%/config/prompt.lua` example:
```
function my_prompt_filter()
    cwd = clink.get_cwd()
    prompt = "\x1b[1;32;40m{cwd}{git}{hg}\x1b[1;30;40m{lamb} \x1b[0m"
    new_value = string.gsub(prompt, "{cwd}", cwd)
    clink.prompt.value = string.gsub(new_value, "{lamb}", "λ")
end

clink.prompt.register_filter(my_prompt_filter, 1)
```
Produces a prompt like:

```
C:\Users\dtgam\cmderdev(user_lua -> origin)λ
```
instead of:

```
C:\Users\dtgam\cmderdev (user_lua -> origin)
λ
```
* Completes dicussion in #1695
* Allows user editing of the prompt for cmd sessions solves or enables a solution for #1425, #1400, #1328, 
* Allows single line prompt #1673

### CODE REVIEW

**Comment by Stanzilla** [comment]
Very nice, especially the prompt stuff, that was requested pretty often, should probably be on the wiki somewhere as well after this has been merged. Have not tested it yet but the code looks fine

**Comment by daxgames** [comment]
I actually wrote the lua stuff months ago.  I've just been having fun this weekend working on stuff that's not work.

**Comment by daxgames** [comment]
I also figured out how to fix the clink prompt defaulting ot origin as a remote when its a local only repo.

Think I'm gonna working the git prompt colors thing next.

File: vendor/init.bat [hunk 1]
```diff
@@ -85,14 +85,18 @@ call "%cmder_root%\vendor\lib\lib_profile"
...
+%lib_console% debug_output init.bat "Env Var - CMDER_ROOT=%CMDER_ROOT%"
+%lib_console% debug_output init.bat "Env Var - debug_output=%debug_output%"
 
 if defined CMDER_USER_CONFIG (
-    %lib_console% debug-output init.bat "CMDER IS ALSO USING INDIVIDUAL USER CONFIG FROM '%CMDER_USER_CONFIG%'!"
+    %lib_console% debug_output init.bat "CMDER IS ALSO USING INDIVIDUAL USER CONFIG FROM '%CMDER_USER_CONFIG%'!"
 )
 
 :: Pick right version of clink
+echo here
```
Comment (DRSDavidSoft) [inline vendor/init.bat]: What's this for?

Comment (daxgames) [reply vendor/init.bat]: DOH!  Should be removed

**Comment by DRSDavidSoft** [comment]
Nice work on fixing #1862, and thanks for completing #1695's discussion!

**Comment by Stanzilla** [comment]
@daxgames how done are you with this PR?

**Comment by daxgames** [comment]
Yes

**Pull request merged**